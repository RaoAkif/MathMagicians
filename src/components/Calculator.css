.calculator {
  border: 1px solid #ccc;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 400px;
}

.calculator-screen {
  width: 100%;
  font-size: 5rem;
  height: 80px;
  border: none;
  background-color: #858694;
  color: #fff;
  text-align: right;
  padding-right: 20px;
  padding-left: 10px;
}

button {
  height: 60px;
  background-color: #e0e0e0;
  border: 1px solid #c4c4c4;
  font-size: 2rem;
  color: #333;
}

button:hover {
  background-color: #eaeaea;
}

.operator {
  color: black;
  background-color: #f5913e;
}

.operator:hover {
  color: black;
  background-color: rgba(245, 144, 61, 0.8);
}

.equal-sign {
  background-color: #f5913e;
  color: black;
}

.equal-sign:hover {
  background-color: #f5913e;
}

.calculator-keys {
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;
  grid-template-rows: 20% 20% 20% 20% 20%;
}

.zero {
  grid-row: 5;
  grid-column: 1/3;
}
