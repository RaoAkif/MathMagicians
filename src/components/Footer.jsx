import React from 'react';

const footerStyle = {
  padding: '5px',
  backgroundColor: 'lightgrey',
  textAlign: 'center',
  fontFamily: 'Calibri',
  fontSize: '12px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  position: 'absolute',
  width: '100vw',
  bottom: '0px',
};

const Header = () => {
  return (
    <div style={footerStyle}>
      Copyright Reserved 2022. Rao Akif
    </div>
  );
};

export default Header;
