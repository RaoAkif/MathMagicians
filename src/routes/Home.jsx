export default function Home() {
  return (
    <main
      style={{
        padding: '1rem 0',
        fontSize: '16px',
        fontFamily: 'Calibri',
        marginTop: '10px',
        marginBottom: '10px',
        marginRight: '20px',
        marginLeft: '20px',
      }}
    >
      <h2>Welcome to the Home Page</h2>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi
        ullam consectetur voluptate quidem doloremque blanditiis, veritatis
        neque asperiores aspernatur quos repellat, deserunt autem. Voluptatem
        quis, atque harum accusamus aperiam eveniet?Lorem, ipsum dolor sit amet
        consectetur adipisicing elit. Excepturi ullam consectetur voluptate
        quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos
        repellat, deserunt autem. Voluptatem quis, atque harum accusamus aperiam
        eveniet?Lorem, ipsum dolor sit amet consectetur adipisicing elit.
        Excepturi ullam consectetur voluptate quidem doloremque blanditiis,
        veritatis neque asperiores aspernatur quos repellat, deserunt autem.
        Voluptatem quis, atque harum accusamus aperiam eveniet?Lorem, ipsum
        dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur
        voluptate quidem doloremque blanditiis, veritatis neque asperiores
        aspernatur quos repellat, deserunt autem. Voluptatem quis, atque harum
        accusamus aperiam eveniet?Lorem, ipsum dolor sit amet consectetur
        adipisicing elit. Excepturi ullam consectetur voluptate quidem
        doloremque blanditiis, veritatis neque asperiores aspernatur quos
        repellat, deserunt autem.
      </p>
    </main>
  );
}
