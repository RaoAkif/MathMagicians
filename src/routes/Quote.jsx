export default function Quote() {
  return (
    <main
      style={{
        padding: '1rem 0',
        fontSize: '16px',
        fontFamily: 'Calibri',
        marginTop: '10px',
        marginBottom: '10px',
        marginRight: '20px',
        marginLeft: '20px',
      }}
    >
      <h2>Quote</h2>
      <p>
        Here is a quote from
        {' '}
        <span style={{ fontWeight: 'bold' }}>
          Muhammad ibn Musa al-Khwarizmi
        </span>
        :
      </p>
      <blockquote cite="https://www.azquotes.com/author/44490-Muhammad_ibn_Musa_al_Khwarizmi#:~:text=Muhammad%20ibn%20Musa%20al%2DKhwarizmi%20Quotes&text=When%20I%20consider%20what%20people,may%20be%20divided%20into%20units.">
        When I consider what people generally want in calculating, I found that
        it always is a number. I also observed that every number is composed of
        units, and that any number may be divided into units. Moreover, I found
        that every number which may be expressed from one to ten, surpasses the
        preceding by one unit: afterwards the ten is doubled or tripled just as
        before the units were: thus arise twenty, thirty, etc. until a hundred:
        then the hundred is doubled and tripled in the same manner as the units
        and the tens, up to a thousand;â€¦ so forth to the utmost limit of
        numeration.
      </blockquote>
    </main>
  );
}
