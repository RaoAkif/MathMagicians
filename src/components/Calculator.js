import React, { Component } from 'react';
import './Calculator.css';
import Result from './Result';

export default class Calculator extends Component {
  render() {
    return (
      <div className="calculator-container">
        <Result />

      </div>
    );
  }
}
