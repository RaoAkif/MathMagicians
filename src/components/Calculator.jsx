import React, { useState, useCallback } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';
import buttonsArray from './buttonsArray';

function Calculator() {
  const [state, setState] = useState({});

  const handleButtonClick = useCallback((event) => {
    setState((prev) => {
      const output = calculate(prev, event.target.value);
      return ({
        ...prev,
        ...output,
      });
    });
  }, []);

  const { total, next, operation } = state;
  const display = (total || '') + (operation || '') + (next || '');

  return (
    <div>
      <h2 style={{ paddingTop: '60px', fontSize: '20px', paddingLeft: '20px' }}>Lets do some Math !!!</h2>
      <div className="calculator">
        <p className="calculator-screen">{display || '0'}</p>
        <div className="calculator-keys">
          {Object.entries(buttonsArray).map(([i, { symbol, value, className }]) => (
            <button onClick={handleButtonClick} key={i} type="button" className={className} value={value}>
              {symbol}
            </button>
          ))}
        </div>
      </div>
    </div>
  );
}

export default Calculator;
