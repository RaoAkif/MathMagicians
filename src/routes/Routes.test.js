import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from './Home';
import Quote from './Quote';
import Footer from '../components/Footer';

describe('Pages renders correctly', () => {
  it('Home page renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Calculator page renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote page renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Footer renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
