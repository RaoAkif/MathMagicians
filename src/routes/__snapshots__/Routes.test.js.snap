// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Calculator page renders correctly 1`] = `
<div>
  <h2
    style={
      Object {
        "fontSize": "20px",
        "paddingLeft": "20px",
        "paddingTop": "60px",
      }
    }
  >
    Lets do some Math !!!
  </h2>
  <div
    className="calculator"
  >
    <p
      className="calculator-screen"
    >
      0
    </p>
    <div
      className="calculator-keys"
    >
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="÷"
      >
        ÷
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="decimal"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="equal-sign"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Pages renders correctly Footer renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgrey",
      "bottom": "0px",
      "display": "flex",
      "fontFamily": "Calibri",
      "fontSize": "12px",
      "justifyContent": "center",
      "padding": "5px",
      "position": "absolute",
      "textAlign": "center",
      "width": "100vw",
    }
  }
>
  Copyright Reserved 2022. Rao Akif
</div>
`;

exports[`Pages renders correctly Home page renders correctly 1`] = `
<main
  style={
    Object {
      "fontFamily": "Calibri",
      "fontSize": "16px",
      "marginBottom": "10px",
      "marginLeft": "20px",
      "marginRight": "20px",
      "marginTop": "10px",
      "padding": "1rem 0",
    }
  }
>
  <h2>
    Welcome to the Home Page
  </h2>
  <p>
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur voluptate quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos repellat, deserunt autem. Voluptatem quis, atque harum accusamus aperiam eveniet?Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur voluptate quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos repellat, deserunt autem. Voluptatem quis, atque harum accusamus aperiam eveniet?Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur voluptate quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos repellat, deserunt autem. Voluptatem quis, atque harum accusamus aperiam eveniet?Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur voluptate quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos repellat, deserunt autem. Voluptatem quis, atque harum accusamus aperiam eveniet?Lorem, ipsum dolor sit amet consectetur adipisicing elit. Excepturi ullam consectetur voluptate quidem doloremque blanditiis, veritatis neque asperiores aspernatur quos repellat, deserunt autem.
  </p>
</main>
`;

exports[`Pages renders correctly Quote page renders correctly 1`] = `
<main
  style={
    Object {
      "fontFamily": "Calibri",
      "fontSize": "16px",
      "marginBottom": "10px",
      "marginLeft": "20px",
      "marginRight": "20px",
      "marginTop": "10px",
      "padding": "1rem 0",
    }
  }
>
  <h2>
    Quote
  </h2>
  <p>
    Here is a quote from
     
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Muhammad ibn Musa al-Khwarizmi
    </span>
    :
  </p>
  <blockquote
    cite="https://www.azquotes.com/author/44490-Muhammad_ibn_Musa_al_Khwarizmi#:~:text=Muhammad%20ibn%20Musa%20al%2DKhwarizmi%20Quotes&text=When%20I%20consider%20what%20people,may%20be%20divided%20into%20units."
  >
    When I consider what people generally want in calculating, I found that it always is a number. I also observed that every number is composed of units, and that any number may be divided into units. Moreover, I found that every number which may be expressed from one to ten, surpasses the preceding by one unit: afterwards the ten is doubled or tripled just as before the units were: thus arise twenty, thirty, etc. until a hundred: then the hundred is doubled and tripled in the same manner as the units and the tens, up to a thousand;… so forth to the utmost limit of numeration.
  </blockquote>
</main>
`;
